src/commonMain/kotlin/fr/acinq/lightning/channel/Channel.kt

```diff
@@ -574,43 +569,51 @@ data class WaitForInit(override val staticParams: StaticParams, override val cur
             event is ChannelEvent.InitFundee -> {
-                val nextState = WaitForOpenChannel(staticParams, currentTip, currentOnChainFeerates, event.temporaryChannelId, event.localParams, event.remoteInit)
+                val nextState = WaitForOpenChannel(staticParams, currentTip, currentOnChainFeerates, event.temporaryChannelId, event.localParams, event.channelConfig, event.remoteInit)
                 Pair(nextState, listOf())
             }
             event is ChannelEvent.InitFunder -> {
-                val fundingPubKey = event.localParams.channelKeys.fundingPubKey
-                val paymentBasepoint = event.localParams.channelKeys.paymentBasepoint
-                val open = OpenChannel(
-                    staticParams.nodeParams.chainHash,
-                    temporaryChannelId = event.temporaryChannelId,
-                    fundingSatoshis = event.fundingAmount,
-                    pushMsat = event.pushAmount,
-                    dustLimitSatoshis = event.localParams.dustLimit,
-                    maxHtlcValueInFlightMsat = event.localParams.maxHtlcValueInFlightMsat,
-                    channelReserveSatoshis = event.localParams.channelReserve,
-                    htlcMinimumMsat = event.localParams.htlcMinimum,
-                    feeratePerKw = event.initialFeerate,
-                    toSelfDelay = event.localParams.toSelfDelay,
-                    maxAcceptedHtlcs = event.localParams.maxAcceptedHtlcs,
-                    fundingPubkey = fundingPubKey,
-                    revocationBasepoint = event.localParams.channelKeys.revocationBasepoint,
-                    paymentBasepoint = paymentBasepoint,
-                    delayedPaymentBasepoint = event.localParams.channelKeys.delayedPaymentBasepoint,
-                    htlcBasepoint = event.localParams.channelKeys.htlcBasepoint,
-                    firstPerCommitmentPoint = keyManager.commitmentPoint(event.localParams.channelKeys.shaSeed, 0),
-                    channelFlags = event.channelFlags,
-                    // In order to allow TLV extensions and keep backwards-compatibility, we include an empty upfront_shutdown_script.
-                    // See https://github.com/lightningnetwork/lightning-rfc/pull/714.
-                    tlvStream = TlvStream(
-                        buildList {
-                            add(ChannelTlv.UpfrontShutdownScript(ByteVector.empty))
-                            if (event.channelVersion.isSet(ChannelVersion.ZERO_RESERVE_BIT)) add(ChannelTlv.ChannelVersionTlv(event.channelVersion))
-                            if (event.channelOrigin != null) add(ChannelTlv.ChannelOriginTlv(event.channelOrigin))
-                        }
-                    )
-                )
-                val nextState = WaitForAcceptChannel(staticParams, currentTip, currentOnChainFeerates, event, open)
-                Pair(nextState, listOf(ChannelAction.Message.Send(open)))
+                when (event.channelType) {
```

Comment:
nit: I find that having more detailed high-level branches is often easier to read than nesting when/ifs...
for example here I would write:
```scala
            event is ChannelEvent.InitFunder && (event.channelType == ChannelType.SupportedChannelType.AnchorOutputs || event.channelType == ChannelType.SupportedChannelType.AnchorOutputsZeroConfZeroReserve) -> ...
``` 
or even define a local `isValidChannelType()` method ?

Reply:
Agreed, it's easy to end up with a lot of nesting in kotlin which can become hard to read.
Done in https://github.com/ACINQ/lightning-kmp/pull/277/commits/914d672f81831a595ae6517dd104ccb31c9172d5